contracts/cw3-dao/src/contract.rs

```diff
@@ -136,22 +132,33 @@ pub fn instantiate(
             // Save gov token
             GOV_TOKEN.save(deps.storage, &cw20_addr.addr())?;
 
-            // Instantiate staking contract with DAO as admin
-            let msg = WasmMsg::Instantiate {
-                code_id: stake_contract_code_id,
-                funds: vec![],
-                admin: Some(env.contract.address.to_string()),
-                label,
-                msg: to_binary(&stake_cw20::msg::InstantiateMsg {
-                    admin: Some(env.contract.address.to_string()),
+            match staking_contract {
+                StakingContractMsg::InstantiateNewStakingContract {
+                    staking_contract_code_id,
                     unstaking_duration,
-                    token_address: cw20_addr.addr().to_string(),
-                })?,
-            };
-
-            let msg = SubMsg::reply_on_success(msg, INSTANTIATE_STAKING_CONTRACT_REPLY_ID);
+                } => {
+                    let msg = WasmMsg::Instantiate {
+                        code_id: staking_contract_code_id,
+                        funds: vec![],
+                        admin: Some(env.contract.address.to_string()),
+                        label,
+                        msg: to_binary(&stake_cw20::msg::InstantiateMsg {
+                            admin: Some(env.contract.address.to_string()),
+                            unstaking_duration,
+                            token_address: cw20_addr.addr().to_string(),
+                        })?,
+                    };
+                    let msg = SubMsg::reply_on_success(msg, INSTANTIATE_STAKING_CONTRACT_REPLY_ID);
 
-            msgs.append(&mut vec![msg]);
+                    msgs.append(&mut vec![msg]);
+                }
+                StakingContractMsg::UseExistingStakingContract { addr } => {
+                    // Validate the address
+                    let staking_contract_addr = deps.api.addr_validate(&addr)?;
```

Comment:
We might want to do a bit more validation here... for example:
- Does this contract's staking token match the cw20 the DAO is configured with?
- Is it a staking contract?

Reply:
This was a concern of mine, what would be the best way to achieve this? Send a query for a known route and fail over safely if it is not present?

Reply:
Could query using the GetConfig route, gives us the token address to then validate.